web/tests/integration/team.test.ts

```diff
@@ -11,3 +18,128 @@ describe("team model", () => {
     const client = await getAPIServiceClient();
   });
 });
+
+describe("user role", () => {
+  test("only authorized roles can update team", async () => {
+    const { rows: teams } = (await integrationDBExecuteQuery(
+      `SELECT id, name FROM "public"."team" limit 1`
+    )) as { rows: Array<{ id: string; name: string }> };
+
+    const nameBeforeUpdate = teams[0].name;
+    const team_id = teams[0].id;
+
+    const { rows: teamMemberships } = (await integrationDBExecuteQuery(
+      `SELECT id, user_id, team_id, role FROM "public"."membership" WHERE "team_id" = '${teams[0].id}'`
+    )) as {
+      rows: Array<{
+        id: string;
+        user_id: string;
+        team_id: string;
+        role: Role_Enum;
+      }>;
+    };
+
+    const mutation = gql`
+      mutation UpdateTeam($id: String!, $name: String!) {
+        update_team(where: { id: { _eq: $id } }, _set: { name: $name }) {
+          affected_rows
+        }
+      }
+    `;
+
+    for (const teamMember of teamMemberships) {
+      const client = await getAPIUserClient({
+        team_id: team_id,
+        user_id: teamMember.user_id,
+      });
+
+      const response = await client.mutate({
+        mutation,
+        variables: {
+          id: team_id,
+          name: "new name",
+        },
+      });
+
+      if (teamMember.role === Role_Enum.Owner) {
+        expect(response.data?.update_team?.affected_rows).toEqual(1);
+
+        // NOTE: revert update changes
+        await integrationDBExecuteQuery(
+          // update team hasura postgresql query
+          `UPDATE "public"."team" SET "name" = 'new name' WHERE "id" = '${team_id}' RETURNING "name";`
+        );
+      } else {
+        expect(response.data?.update_team?.affected_rows).toEqual(0);
+      }
+    }
+  });
+
+  test("cannot select another team", async () => {
```

Comment:
**blocking**. one important case we're not covering here because this is calling Hasura GraphQL directly is using the `/v1/graphql` endpoint from Next.js, particularly because anyone can tamper with the team ID header

Reply:
Alternatively the test already checks that a user not part of the team cannot update